src/ivc/step_folding_circuit.rs

```diff
@@ -106,13 +106,15 @@ where
     C: CurveAffine,
     RO: ROCircuitTrait<C::Base>,
 {
-    pub fn without_witness<const A2: usize, SC: StepCircuit<A2, C::Scalar>>(
+    pub fn without_witness<PairedCircuit: Circuit<C::Scalar>>(
         k_table_size: u32,
         num_io: usize,
         step_pp: &'link StepParams<C::Base, RO>,
     ) -> Self {
         let mut cs = ConstraintSystem::<C::Scalar>::default();
-        SC::configure(&mut cs);
+
+        PairedCircuit::configure(&mut cs);
```

Comment:
what does PairedCircuit stand for here?

Reply:
IIUC, this line is the fix

Reply:
Well, in the source code you had two errors here that I didn't notice because of the PR size:

1. To configure primary, `configurate` from primary was used, but in primary we put secondary trace, so here we need `secondary::configurate`. And vice versa respectively.

2. Inside `without_witness` was configured not `StepFoldingCircuit` but just `StepCircuit`, which for example for trivial led to zero in `folding_degree` and was generally wrong. 

I noticed and corrected the first error earlier, the second one only now.